### PR TITLE
fix: silence PollingBlockTracker Sentry error emit

### LINKED ISSUE #7579: [Sentry] Error: PollingBlockTracker - encountered an error while attempting to update latest block:
Sentry Issue: [METAMASK-MOBILE-23WY](https://metamask.sentry.io/issues/4521376341/?referrer=github_integration)

```
Error: PollingBlockTracker - encountered an error while attempting to update latest block:
TypeError: Network request failed
    at anonymous (address at index.android.bundle:1:378491)
    at apply (native)
    at anonymous (address at index.android.bundle:...
  at anonymous (app:///index.android.bundle:1:378491)
  at anonymous (app:///index.android.bundle:1:346857)
  at _callTimer (app:///index.android.bundle:1:346139)
  at callTimers (app:///index.android.bundle:1:347941)
  at __callFunction (app:///index.android.bundle:1:251161)
...
(21 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by frankvonhoven**
## **Description**
1. **What is the reason for the change?**
  * Sentry PollingBlockTracker errors were very noisy (see related issue)
2. **What is the improvement/solution?**
  * Silencing the unhandled error as we weren't able to determine sources of errors
  * Patch the web3-provider-engine node module with change
  * The PollingBlockTracker will be replaced in the network-controller v9 - which is expected to be released soon

## **Related issues**

Fixes: #7579 

## **Manual testing steps**

1. Switch between networks
2. Swap some crypto
3. Send some crypto
4. Shouldn't see any PollingBlockTracker errors like in the related issue in Sentry

### Switch Networks

https://github.com/MetaMask/metamask-mobile/assets/141057783/c9a595cf-1ad2-4b8e-b6ec-381a2cf945ed

### Swap

https://github.com/MetaMask/metamask-mobile/assets/141057783/0daa4aff-fd15-496a-aebd-8c4d9580c23a

### Send

https://github.com/MetaMask/metamask-mobile/assets/141057783/6816e057-ce8a-43d7-abdd-5f8d74169fd4

### Transaction History

https://github.com/MetaMask/metamask-mobile/assets/141057783/bf1fe57c-bf2c-4d9e-aa4a-bbe0aab8e4bb

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: patches/web3-provider-engine+16.0.5.patch [hunk 1]
```diff
@@ -1,7 +1,121 @@
...
+-const { PollingBlockTracker } = require('eth-block-tracker')
++const { PollingBlockTracker } = require('./PollingBlockTracker')
+ const map = require('async/map')
+ const eachSeries = require('async/eachSeries')
+ const Stoplight = require('./util/stoplight.js')
+@@ -24,7 +24,7 @@ function Web3ProviderEngine(opts) {
+   // block polling
+   const directProvider = { sendAsync: self._handleAsync.bind(self) }
+   const blockTrackerProvider = opts.blockTrackerProvider || directProvider
+-  self._blockTracker = opts.blockTracker || new PollingBlockTracker({
```
Comment (Cal-L) [inline patches/web3-provider-engine+16.0.5.patch:114]: Remove this line change

**Review by Cal-L — COMMENTED** [review]
Left a comment

**Comment by Cal-L** [comment]
Could we also provide evidence showing that these aren't broken since PollingBlockTracker is used by both network controller and transaction controller?

- Switching networks between testnets as well as other chains such as BNB, polygon, etc
- Ensuring that transaction history still appears properly in the activity view
- Perform some transactions on mainnet such as send, swap, etc

**Review by Cal-L — APPROVED** [review]
LGTM

**Comment by chrisleewilcox** [comment]
@frankvonhoven @Cal-L I moved this back to dev review.  E2E failed to build.  Can you update branch and rerun E2E for this PR?

**Review by Cal-L — APPROVED** [review]
Lgtm

**Review by chrisleewilcox — APPROVED** [review]
LGTM

**Pull request merged**